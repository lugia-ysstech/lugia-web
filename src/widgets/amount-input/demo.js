/**
 *
 * create by ligx
 *
 * @flow
 */
import * as React from 'react';
import Theme from '../theme';
import AmountInput from './index';
import Widget from '../consts/index';
import styled from 'styled-components';
import { getBoxShadow } from '@lugia/theme-utils';

class LimitAmountInput extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = { value: '' };
  }

  onChange = ({ newValue: value }: any) => {
    this.setState({ value });
    this.props.onChange({ newValue: value });
  };

  render() {
    return <AmountInput value={this.state.value} onChange={this.onChange} />;
  }
}
class DefaultValueAmountInput extends React.Component<any, any> {
  render() {
    return <AmountInput defaultValue={'123456'} onChange={this.props.onChange} />;
  }
}
const Wrapper = styled.div`
  float: left;
  margin-left: 50px;
`;
export default () => {
  const view = {
    [Widget.AmountInput]: {
      Container: {
        normal: {
          width: 400,
          height: 40,
          border: {
            top: 10,
          },
        },
      },
      AmountInputPrefix: { normal: { fontSize: 14, color: 'blue' } },
      AmountTip: {
        Container: {
          normal: {
            background: {
              color: '#eee',
            },
          },
        },
        TooltipTitle: {
          normal: {
            color: '#4d63ff',
            fontSize: 16,
          },
        },
      },
    },
  };
  const onChange = (cmpName: string) => (value: any) => {};
  return (
    <div>
      <Wrapper>
        <p>formatter AmountInput</p>
        <AmountInput placeholder={'请填写金额'} />
        <p>default AmountInput</p>
        <AmountInput placeholder={'请填写金额'} />
        <p>禁用状态 </p>
        <AmountInput size={'default'} disabled={true} />
      </Wrapper>
      <Wrapper>
        <Theme config={view}>
          <p>small size</p>
          <AmountInput size={'small'} placeholder={'请填写金额'} />
          <p>default size</p>
          <AmountInput placeholder={'请填写金额'} />
          <p>large size</p>
          <AmountInput size={'large'} placeholder={'请填写金额'} disabled={true} />
        </Theme>
      </Wrapper>
      <Wrapper>
        <p>受限Input</p>
        <LimitAmountInput onChange={onChange('limit')} />
        <p>有默认值的 受限Input</p>
        <DefaultValueAmountInput value={'123456'} onChange={onChange('limit')} />
      </Wrapper>
      <Wrapper>
        <p>amountPrefix: '¥' transform: false </p>
        <AmountInput amountPrefix="¥" transform={false} />
        <p>amountPrefix: '$' transform: false </p>
        <AmountInput amountPrefix="$" transform={false} />
      </Wrapper>
    </div>
  );
};
