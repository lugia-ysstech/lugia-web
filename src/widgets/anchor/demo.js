/**
 *
 * create by guorg
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';
import Theme from '../theme';
import Widget from '../consts/index';
import Anchor from './index';

const { Link } = Anchor;
const DemoBox = styled.div`
  padding: 20px;
  margin-left: ${props => 120 + props.cur * 120}px;
  width: 209px;
  height: 203px;
`;

export default class DrawerDemo extends React.Component<any, any> {
  constructor() {
    super();
  }
  click = () => {
    const anchorElement = document.getElementById('anchor-link');
    if (anchorElement) {
      anchorElement.scrollIntoView();
    }
  };

  handleLinkClick = (e, href) => {
    if (href) {
      const name = href.slice(1);
      const anchorElement = document.getElementById(name);
      if (anchorElement) {
        anchorElement.scrollIntoView({ block: 'start', behavior: 'smooth' });
      }
    }
  };
  render() {
    return (
      <div>
        <DemoBox cur={0}>
          <Anchor slideType="circle" useHref={false} onClick={this.handleLinkClick}>
            <Link title="anchor-link1" href="#anchor-link1" />
            <Link title="anchor-link2" href="#anchor-link2" />
            <Link title="anchor-link3" href="#anchor-link3">
              <Link title="anchor-link3.1" href="#anchor-link3.1" />
              <Link title="anchor-link3.2" href="#anchor-link3.2" />
            </Link>
            <Link title="anchor-link4" href="#anchor-link4" />
          </Anchor>
        </DemoBox>
        <DemoBox cur={1}>
          <Anchor slideType="line">
            <Link title="anchor-link1" href="#anchor-link1" />
            <Link title="anchor-link2" href="#anchor-link2" />
            <Link title="anchor-link3" href="#anchor-link3">
              <Link title="anchor-link3.1" href="#anchor-link3.1" />
              <Link title="anchor-link3.2" href="#anchor-link3.2" />
            </Link>
            <Link title="anchor-link4" href="#anchor-link4" />
          </Anchor>
        </DemoBox>
        <DemoBox cur={2}>
          <Anchor slideType="none" affix={false}>
            <Link title="anchor-link1" href="#anchor-link1" />
            <Link title="anchor-link2" href="#anchor-link2" />
            <Link title="anchor-link3" href="#anchor-link3">
              <Link title="anchor-link3.1" href="#anchor-link3.1" />
              <Link title="anchor-link3.2" href="#anchor-link3.2" />
            </Link>
            <Link title="anchor-link4" href="#anchor-link4" />
          </Anchor>
        </DemoBox>
        <div style={{ marginTop: '100px' }} id="anchor-link1" name="anchor-link1">
          anchor-link1
        </div>
        <div style={{ marginTop: '300px' }} id="anchor-link2" name="anchor-link2">
          anchor-link2
        </div>
        <div style={{ marginTop: '300px' }} id="anchor-link3" name="anchor-link3">
          anchor-link3
        </div>
        <div style={{ marginTop: '200px' }} id="anchor-link3.1" name="anchor-link3.1">
          anchor-link3.1
        </div>
        <div style={{ marginTop: '200px' }} id="anchor-link3.2" name="anchor-link3.2">
          anchor-link3.2
        </div>
        <div style={{ marginTop: '300px' }} id="anchor-link4" name="anchor-link4">
          anchor-link4
        </div>
      </div>
    );
  }
}
