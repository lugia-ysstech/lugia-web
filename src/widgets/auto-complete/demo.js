import * as React from 'react';
import AutoComplete from './';
import { getBorder } from '@lugia/theme-utils';
import Widget from '../consts/index';
import { getBorderRadius } from '../theme/CSSProvider';

const data = [
  'Nikcy Romero',
  'Armin van Buuren',
  'Hardwell',
  'Zedd',
  'Kazze',

  'Vicetone',
  'Martin Garrix',
  'David Guetta',
  'The Chainsmokers',
  'Kygo',

  'Axwell ^ Ingrosso',
  'Dimitri Vegas & Like Mike',
  'Calvin Harris',
  'Avicci',
  'Fedde Le Grand',

  'Tiesto',
  'Snoop Dogg',
  'Bassjackers',
  'Sebastian Ingrosso',
  'Swedish House Mafia',
  'Alesso',
  'Afrojack',
  'Knife Party',
  'Dannic',
  'R3hab',
];

export default class AutoCompleteBounded extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      menuData: data,
    };
  }
  static getDerivedStateFromProps(props: any, state: any) {
    const hasValueInProps = 'value' in props;
    const value = hasValueInProps ? props.value : state ? state.value : props.defaultValue;
    if (!state) {
      return {
        value,
      };
    }
    return {
      value,
    };
  }

  render() {
    const config = {
      [Widget.AutoComplete]: {
        OldItem: {
          normal: {
            height: 50,
            color: '#000',
            font: { size: 20 },
            background: { color: '#333' },
            opacity: 0.6,
            padding: { left: 10 },
            border: getBorder({ color: '#4d63ff', width: 1, style: 'solid' }),
            borderRadius: getBorderRadius(20),
          },
          hover: {
            color: '#4d63ff',
            background: { color: '#4a8e29' },
            opacity: 1,
            border: getBorder({ color: '#4d63ff', width: 1, style: 'solid' }),
            borderRadius: getBorderRadius(20),
          },
        },

        Container: {
          normal: {
            width: 300,
            height: 60,
          },
        },

        Menu: {
          MenuWrap: {
            normal: {
              width: 300,
              height: 350,
              opacity: 0.6,
              boxShadow: '2px 2px 5px #4d63ff',
              background: { color: '#000' },
              border: getBorder({ color: '#4d63ff', width: 1, style: 'solid' }),
              borderRadius: getBorderRadius(20),
            },
            hover: {
              opacity: 1,
            },
          },
          MenuItem: {
            normal: { color: '#ccc', fontSize: 14, font: { fontWeight: 900 } },
            hover: {
              color: '#fff',
              fontSize: 20,
              background: { color: 'green' },
              font: { fontWeight: 400 },
            },
            active: {
              color: 'blue',
              fontSize: 14,
              background: { color: 'pink' },
              font: { fontWeight: 900 },
            },
            disabled: { color: 'red', background: { color: '#000' } },
          },
          SelectedMenuItem: {
            normal: {
              color: 'blue',
              font: { fontWeight: 900 },
              fontSize: 18,
              background: { color: 'orange' },
            },
            hover: { color: '#000', background: { color: 'yellow' } },
            active: { color: 'green' },
          },
          Divider: { normal: { color: 'red' } },
        },
      },
    };
    const { menuData, value } = this.state;
    return (
      <AutoComplete
        theme={config}
        showOldValue
        value={value}
        data={menuData}
        onChange={this.onChange}
      />
    );
  }

  onChange = (value: string) => {
    this.search(value);
    this.setState({ value });
  };

  search(query: string) {
    let menuData;
    let rowSet = [];
    const len = data.length;

    for (let i = 0; i < len; i++) {
      const row = data[i];
      if (this.searchValue(query, row)) {
        rowSet.push(row);
      }
      if (query === row) {
        rowSet = [];
        break;
      }
    }

    if (rowSet.length === len) {
      menuData = data;
    } else {
      menuData = rowSet;
    }
    this.setState({ menuData });
  }

  searchValue = (query: string, row: string): boolean => {
    return row.indexOf(query) !== -1;
  };
}

// export default class Demo extends React.Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       menuData: data,
//       value: '',
//     };
//   }
//   render() {
//     return (
//       <AutoComplete
//         data={this.state.menuData}
//         onChange={this.onChange}
//         placeholder={'请输入'}
//         showOldValue={true}
//       />
//     );
//   }

//   onChange = value => {
//     this.search(value);
//   };

//   search(query) {
//     let menuData;
//     let rowSet = [];
//     const len = data.length;

//     for (let i = 0; i < len; i++) {
//       const row = data[i];
//       if (query === row) {
//         rowSet = [];
//         break;
//       }

//       if (this.searchValue(query, row)) {
//         rowSet.push(row);
//       }
//     }

//     if (rowSet.length === len) {
//       menuData = data;
//     } else {
//       menuData = rowSet;
//     }
//     this.setState({ menuData });
//   }

//   searchValue = (query, row) => {
//     return row.indexOf(query) !== -1 || row === query;
//   };
// }
