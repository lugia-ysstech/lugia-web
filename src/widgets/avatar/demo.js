/**
 *
 * create by liangguodong on 2018/8/27
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Avatar from './';
import Widget from '../consts';
import Theme from '../theme';
import Badge from '../badge/index';

const Wrapper = styled.div`
  float: left;
  margin-left: 50px;
  text-align: center;
  padding: 20px;
`;
export default () => {
  const badgeConfig = {
    [Widget.Badge]: {
      BadgeDot: {
        normal: {
          position: { right: 0, top: -3 },
        },
      },
      BadgeNumber: {
        InnerNumber: {
          normal: { position: { right: -5, top: -3 } },
        },
      },
    },
  };
  const config = {
    [Widget.Avatar]: {
      Container: { normal: { background: { color: '#e2e2e2' } } },
      IconAvatar: { normal: { color: 'orange' } },
      SrcAvatar: { normal: { width: 24, height: 24 } },
      FontAvatar: { normal: { color: 'red' } },
    },
  };
  const iconConfig = {
    [Widget.Avatar]: {
      IconAvatar: { normal: { color: 'purple', fontSize: 16 } },
    },
  };
  return (
    <div>
      <Theme config={config}>
        <Wrapper>
          <Avatar shape={'square'} name={'l'} />
          <Avatar shape={'square'} name={'lu'} />
          <Avatar viewClass="config" shape={'square'} name={'lug'} />
          <Avatar viewClass="config" shape={'square'} name={'lugia'} />
          <br />
          <Avatar shape={'circle'} name={'l'} />
          <Avatar shape={'circle'} name={'lu'} />
          <Avatar viewClass="config" shape={'circle'} name={'lug'} />
          <Avatar viewClass="config" shape={'circle'} name={'lugia'} />
        </Wrapper>
        <Wrapper>
          <Theme config={iconConfig}>
            <Avatar
              shape={'square'}
              type="icon"
              icon={'lugia-icon-financial_user'}
              size={'small'}
            />
            <Avatar
              shape={'square'}
              type="icon"
              icon={'lugia-icon-financial_user'}
              size={'default'}
            />
            <Avatar
              shape={'square'}
              type="icon"
              icon={'lugia-icon-financial_user'}
              size={'large'}
            />
          </Theme>
          <br />
          <Avatar shape={'circle'} type="icon" icon={'lugia-icon-financial_user'} size={'small'} />
          <Avatar
            shape={'circle'}
            type="icon"
            icon={'lugia-icon-financial_user'}
            size={'default'}
          />
          <Avatar shape={'circle'} type="icon" icon={'lugia-icon-financial_user'} size={'large'} />
        </Wrapper>
        <Wrapper>
          <Theme
            config={{
              register: {
                FontAvatar: { normal: { color: 'blue' } },
              },
            }}
          >
            <Avatar viewClass="register" shape={'square'} size={'small'} name={'lugia'} />
            <Avatar viewClass="register" shape={'square'} size={'default'} name={'lugia'} />
            <Avatar viewClass="register" shape={'square'} size={'large'} name={'lugia'} />
          </Theme>
          <br />
          <Avatar shape={'circle'} size={'small'} name={'lugia'} />
          <Avatar shape={'circle'} size={'default'} name={'lugia'} />
          <Avatar shape={'circle'} size={'large'} name={'lugia'} />
        </Wrapper>
        <Wrapper>
          <Avatar shape={'square'} name={'lugia'} msgNum={1} />
          <Avatar shape={'square'} type="icon" icon={'lugia-icon-financial_user'} msgNum={2} />
          <Avatar
            shape={'square'}
            type="img"
            src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/master/lugiaweb%E7%BB%84%E4%BB%B6/%E5%A4%B4%E5%83%8F/32.jpg"
            msgNum={99}
          />
          <br />
          <Avatar shape={'circle'} name={'lugia'} />
          <Avatar shape={'circle'} type="icon" icon={'lugia-icon-financial_user'} />
          <Avatar
            shape={'circle'}
            type="img"
            src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/master/lugiaweb%E7%BB%84%E4%BB%B6/%E5%A4%B4%E5%83%8F/32.jpg"
            msgNum={99}
          />
        </Wrapper>
      </Theme>

      <Wrapper>
        <Theme config={badgeConfig}>
          <Badge count={10}>
            <Avatar shape={'square'} size={'large'} name={'lugia'} />
          </Badge>
          <br />
          <br />
          <Badge dot={true}>
            <Avatar shape={'square'} size={'large'} name={'lugia'} />
          </Badge>
        </Theme>
        <br />
        <br />
      </Wrapper>
      <Wrapper>
        <Theme config={badgeConfig}>
          <Badge viewClass="badgeConfig" count={10}>
            <Avatar icon={'lugia-icon-financial_user'} size={'large'} type="icon" />
          </Badge>
          <br />
          <br />
          <Badge viewClass="badgeConfig" dot={true}>
            <Avatar icon={'lugia-icon-financial_user'} size={'large'} type="icon" />
          </Badge>
        </Theme>
      </Wrapper>
    </div>
  );
};
