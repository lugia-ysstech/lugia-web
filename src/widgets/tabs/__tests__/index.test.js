/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Tabs, { hasTargetInProps, getDefaultData } from '../tabs';
import { hasActivityValueData, defaultData, shortChildren, longChildren } from '../demo';

Enzyme.configure({ adapter: new Adapter() });

describe('tabsDemo', () => {
  const themeProps = { themeConfig: {}, themeState: {} };
  const getPartOfThemeProps = () => true;
  const createTabs = (obj?: Object): any => {
    const config = obj || {
      activityValue: 0,
      tabType: 'line',
      tabPosition: 'bottom',
      data: hasActivityValueData,
    };
    return (
      <Tabs
        {...config}
        getPartOfThemeHocProps={getPartOfThemeProps}
        getPartOfThemeProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );
  };
  it('Component JSON', () => {
    const target = (
      <Tabs
        getPartOfThemeProps={getPartOfThemeProps}
        getPartOfThemeHocProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );
    expect(renderer.create(target).toJSON()).toMatchSnapshot();
  });
  const getCmp = (target: any): Object => {
    return target.children().instance();
  };

  it('props hasTargetInProps', () => {
    expect(hasTargetInProps('data', { data: [1, 2, 3] })).toBe(true);
    expect(hasTargetInProps('children', { children: [1, 2, 3] })).toBe(true);
    expect(hasTargetInProps('ops', { ops: 'sdfsdf' })).toBe(true);
    expect(hasTargetInProps('ads', { ops: 'sdfsdf' })).toBe(false);
  });

  it('props defaultActivityValue', () => {
    const target = mount(createTabs({ defaultActivityValue: '2', data: hasActivityValueData }));
    const { activityValue } = getCmp(target).state;
    expect(activityValue).toBe('2');
  });
  it('props activityValue', () => {
    const target = mount(createTabs({ activityValue: '2', data: hasActivityValueData }));
    const { activityValue } = getCmp(target).state;
    expect(activityValue).toBe('2');
  });
  it('props defaultData', () => {
    const target = mount(createTabs({ defaultData }));
    const { data } = getCmp(target).state;
    expect(data).toEqual(data);
  });
  it('props data', () => {
    const target = mount(createTabs({ data: hasActivityValueData, tabPosition: 'left' }));
    const { data } = getCmp(target).state;
    expect(data).toEqual(hasActivityValueData);
  });
  it('props data && defaultData', () => {
    const target = mount(createTabs({ data: hasActivityValueData, defaultData }));
    const { data } = getCmp(target).state;
    expect(data).toEqual(hasActivityValueData);
  });
  it('props data []', () => {
    const target = mount(createTabs({ data: [] }));
    const { data } = getCmp(target).state;
    expect(data).toEqual([]);
  });
  it('props children shortChildren', () => {
    const target = mount(createTabs({ children: shortChildren }));
    const { activityValue } = getCmp(target).state;
    expect(activityValue).toBe('Tab1');
  });
  it('props children longChildren', () => {
    const target = mount(createTabs({ children: longChildren }));
    const { activityValue } = getCmp(target).state;
    expect(activityValue).toBe('Tab1');
  });
  it('props children&&data', () => {
    const target = mount(createTabs({ data: hasActivityValueData, children: shortChildren }));
    const { activityValue } = getCmp(target).state;
    expect(activityValue).toBe('0');
  });

  function testActivityValue(component: any, expActicityValue: string) {
    it('props activityValue', () => {
      const target = mount(component);
      const type = 'Tabpane';
      const tabpane = target.find(type);
      tabpane.at(2).simulate('click');
      expect(getCmp(target).state.activityValue).toBe(expActicityValue);
    });
  }
  testActivityValue(createTabs({ data: hasActivityValueData, activityValue: '1' }), '1');
  testActivityValue(createTabs({ data: hasActivityValueData, activityValue: '3' }), '3');

  it('props onAddClick 非受限', async () => {
    const target = mount(
      <Tabs
        tabType="card"
        getPartOfThemeHocProps={getPartOfThemeProps}
        getPartOfThemeProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );
    expect(getCmp(target).state.data).toEqual([
      { title: 'Tab1', content: 'Tab1 content', key: 'Tab1' },
      { title: 'Tab2', content: 'Tab2 content', key: 'Tab2' },
      { title: 'Tab3', content: 'Tab3 content', key: 'Tab3' },
    ]);
    target
      .children()
      .instance()
      .onAddClick();
    expect(getCmp(target).state.data).toEqual([
      { title: 'Tab1', content: 'Tab1 content', key: 'Tab1' },
      { title: 'Tab2', content: 'Tab2 content', key: 'Tab2' },
      { title: 'Tab3', content: 'Tab3 content', key: 'Tab3' },
      { title: 'Tab4', content: 'Tab4 Content', key: 'Tab4' },
    ]);
  });

  it('props onAddClick 受限', async () => {
    const target = mount(createTabs());
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
    target
      .children()
      .instance()
      .onAddClick();
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
  });

  it('props onDelete 非受限', async () => {
    const target = mount(
      <Tabs
        tabType="card"
        getPartOfThemeHocProps={getPartOfThemeProps}
        getPartOfThemeProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );
    expect(getCmp(target).state.data).toEqual([
      { title: 'Tab1', content: 'Tab1 content', key: 'Tab1' },
      { title: 'Tab2', content: 'Tab2 content', key: 'Tab2' },
      { title: 'Tab3', content: 'Tab3 content', key: 'Tab3' },
    ]);
    target
      .children()
      .instance()
      .onDelete({ index: 1, activityValue: 'Tab2' });
    expect(getCmp(target).state.data).toEqual([
      { title: 'Tab1', content: 'Tab1 content', key: 'Tab1' },
      { title: 'Tab3', content: 'Tab3 content', key: 'Tab3' },
    ]);
  });

  it('props onDelete 受限', async () => {
    const target = mount(createTabs());
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
    target
      .children()
      .instance()
      .onDelete(2);
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
  });

  it('props onChange 非受限', async () => {
    const target = mount(
      <Tabs
        tabType="card"
        getPartOfThemeHocProps={getPartOfThemeProps}
        getPartOfThemeProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );
    expect(getCmp(target).state.activityValue).toEqual('Tab1');
    target
      .children()
      .instance()
      .onChange({ index: 1, activityValue: 'Tab2' });
    expect(getCmp(target).state.activityValue).toEqual('Tab2');
  });

  it('props onChange 受限', async () => {
    const target = mount(createTabs());
    expect(getCmp(target).state.activityValue).toEqual('0');
    target
      .children()
      .instance()
      .onChange({ index: 2, activityValue: 'Tab3' });
    expect(getCmp(target).state.activityValue).toEqual('0');
  });

  it('props onAddClick 非受限', async () => {
    const target = mount(
      <Tabs
        tabType="card"
        getPartOfThemeHocProps={getPartOfThemeProps}
        getPartOfThemeProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );

    expect(getCmp(target).state.data).toEqual(getDefaultData({}));
    expect(getCmp(target).state.activityValue).toEqual('Tab1');
    target
      .children()
      .instance()
      .onAddClick();
    expect(getCmp(target).state.data).toEqual([
      { title: 'Tab1', content: 'Tab1 content', key: 'Tab1' },
      { title: 'Tab2', content: 'Tab2 content', key: 'Tab2' },
      { title: 'Tab3', content: 'Tab3 content', key: 'Tab3' },
      { title: 'Tab4', content: 'Tab4 Content', key: 'Tab4' },
    ]);
    expect(getCmp(target).state.activityValue).toEqual('Tab4');
  });

  it('props onAddClick 受限', async () => {
    const target = mount(createTabs());
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
    expect(getCmp(target).state.activityValue).toEqual('0');
    target
      .children()
      .instance()
      .onAddClick();
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
    expect(getCmp(target).state.activityValue).toEqual('0');
  });

  it('props 受限 data', async () => {
    const target = mount(
      <Tabs
        data={hasActivityValueData}
        getPartOfThemeHocProps={getPartOfThemeProps}
        getPartOfThemeProps={getPartOfThemeProps}
        themeProps={themeProps}
      />
    );
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
    getCmp(target).setState({ data: [] });
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
    target.setProps({ data: [] });
    expect(getCmp(target).state.data).toEqual([]);
    target.setProps({ data: hasActivityValueData });
    expect(getCmp(target).state.data).toEqual(hasActivityValueData);
  });

  it('getCurrentPageByActivityValue', async () => {
    const target = mount(createTabs());
    const mockInfo = [
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '1',
        totalPage: 4,
        exception: 0,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '2',
        totalPage: 4,
        exception: 0,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '3',
        totalPage: 4,
        exception: 0,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '4',
        totalPage: 4,
        exception: 1,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '5',
        totalPage: 4,
        exception: 1,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '6',
        totalPage: 4,
        exception: 1,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '7',
        totalPage: 4,
        exception: 2,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '8',
        totalPage: 4,
        exception: 2,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '9',
        totalPage: 4,
        exception: 2,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
        ],
        activityValue: '10',
        totalPage: 4,
        exception: 3,
      },
      {
        data: [
          { key: '1' },
          { key: '2' },
          { key: '3' },
          { key: '4' },
          { key: '5' },
          { key: '6' },
          { key: '7' },
          { key: '8' },
          { key: '9' },
          { key: '10' },
          { key: '11' },
          { key: '12' },
          { key: '13' },
          { key: '14' },
          { key: '15' },
        ],
        activityValue: '7',
        totalPage: 4,
        exception: 1,
      },
    ];
    mockInfo.forEach(item => {
      const { exception, data, activityValue, totalPage } = item;
      const tabHeader = target.find('TabHeader');
      const res = tabHeader
        .instance()
        .getCurrentPageByActivityValue(data, activityValue, totalPage);
      expect(res).toEqual(exception);
    });
  });
});
