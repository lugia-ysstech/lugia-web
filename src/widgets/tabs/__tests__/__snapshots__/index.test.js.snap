// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabsDemo Component JSON 1`] = `
.c4 {
  position: relative;
  cursor: pointer;
  white-space: nowrap;
  padding: 0 20px;
  display: inline-block;
  text-align: center;
}

.c4 > div::before {
  content: '';
  background: #4d63ff;
  border-radius: 2px;
  position: absolute;
  bottom: 0px;
  width: 100%;
  height: 2px;
  left: 50%;
  -webkit-animation: gaMprO 0.2s linear forwards;
  animation: gaMprO 0.2s linear forwards;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c6 {
  position: relative;
  cursor: pointer;
  white-space: nowrap;
  padding: 0 20px;
  display: inline-block;
  text-align: center;
}

.c5 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 4.2rem;
  height: 4.2rem;
}

.c5 > * {
  max-width: 100%;
}

.c2 {
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  float: left;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  min-width: 100%;
  -webkit-transform: translateX(0rem);
  -ms-transform: translateX(0rem);
  transform: translateX(0rem);
}

.c1 {
  position: relative;
  z-index: 99;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
}

.c1::before {
  content: '';
  position: absolute;
  bottom: 0;
  height: 1px;
  background-color: #e8e8e8;
  left: 0;
  width: 100%;
}

.c7 {
  position: relative;
  background: #fff;
  min-width: 100px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
}

.c0 {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  position: relative;
  min-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="OutContainer  c0"
  style={Object {}}
>
  <div
    className="c1"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  >
    <div
      className="c2"
      style={Object {}}
    >
      <div
        className="c3"
        style={Object {}}
        x={-0}
      >
        <div
          className="c4"
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#4d63ff",
            }
          }
        >
          <div
            className="lineTitle  c5"
            disabled={false}
            style={
              Object {
                "lineHeight": "3.4rem",
              }
            }
          >
            Tab1
          </div>
        </div>
        <div
          className="c6"
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={Object {}}
        >
          <div
            className="lineTitle  c5"
            disabled={false}
            style={
              Object {
                "lineHeight": "3.4rem",
              }
            }
          >
            Tab2
          </div>
        </div>
        <div
          className="c6"
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={Object {}}
        >
          <div
            className="lineTitle  c5"
            disabled={false}
            style={
              Object {
                "lineHeight": "3.4rem",
              }
            }
          >
            Tab3
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c7"
    style={Object {}}
  />
</div>
`;
